pyiron/sphinx/base.py

```diff
@@ -1161,11 +1134,24 @@ def write_group(self, group, file_name):
         with open(file_name, "w") as f:
             f.write(group.to_sx_str())
 
-    def write_input(self):
+    def write_input(self, file_name="input.sx"):
         """
         Generate all the required input files for the Sphinx job.
 
+        Creates:
+        userparameters.sx: user-defined variables in job.input
```

Comment:
Would it hurt to write the user parameters directly into input.sx? Many important settings are in here, that are important in different places in the input file later on, so having a chance to just scroll up to see what they are would be helpful for me.

Reply:
It's maybe my personal thing, but I almost **never** open SPHInX input files. So whether things are written in different files or not hardly affects how I work.

Reply:
> It's maybe my personal thing, but I almost **never** open SPHInX input files.

Is that why I get an email when SPHInX doesn't work for you ;-) ? - I don't insist on this, but having the content of userparameters.sx inside the input.sx (or maybe even putting settings right where they belong) would render **my** life easier when I am asked why SPHInX doesn't do what the pyiron users expected it to do.

Reply:
I like this suggestion, but is there a good reason to write these as variables to the input file at all anymore? If everything is in one file, it seems more elegant to me just to write their values straight to the places they are used

Reply:
> Is that why I get an email when SPHInX doesn't work for you ;-) ?

Yeah and if it doesn't work, then it usually doesn't work anyway, whether I look at the input files or not :)

> I like this suggestion, but is there a good reason to write these as variables to the input file at all anymore? If everything is in one file, it seems more elegant to me just to write their values straight to the places they are used

That's true. I guess as long as `to_sx_str` is available, I don't think we need `user parameters.sx`.